### ISSUE TITLE
Request: Starting AppDriver in a custom URL for testing bookmarking

### ISSUE THREAD

**Issue by zsigmas** [rstudio/shinytest2#219]
Hi all, I have not yet found a direct way of testing bookmarking. This is relevant when creating modules that you want to guarantee will save and restore bookmarks correctly.

One possibility would be to include a parameter in the `AppDriver$new` method:
```
app <- appDriver$new(appObject, query_args = "/?inputs_&n=10"
```

**Comment by zsigmas** [comment]
Thanks a lot!

**Comment by zsigmas** [comment]
Maybe a simpler solution for this problem could be:

(notice I have corrected some typos in the URL from the previous response)

```
bg_app <- shinytest2::AppDriver$new(app_dir = "my/shiny/app")
bg_app_url <- bg_app$get_url()
app<- shinytest2::AppDriver$new(paste0(bg_app_url, "/?_inputs_&n=10"))
```

This delegates all the app launching process to `shinytest2::AppDriver$new`

Unfortunately, this has the following problem:
```
bg_app <- shinytest2::AppDriver$new(app_dir = "my/shiny/app")
bg_app_url <- bg_app$get_url() # Assume that bg_app_url is 127.0.0.1:1234
app<- shinytest2::AppDriver$new(paste0(bg_app_url, "/?inputs_n=10"))
app$get_url(shinytest2::AppDriver$new(paste0(bg_app_url, "/?_inputs_&n=10"))
# [1] http://127.0.0.1:1234/?inputs=&n=10
```

I was able to pinpoint the problem to:
https://github.com/rstudio/shinytest2/blob/29c7a6b2cfc83de720acc5ca86cdab73dca7ad0e/R/R6-helper.R#L60

I will report this in a different issue that links here to separate it from this enhancement request.

**Comment by schloerke** [comment]
What a wonderful solution!

```r
bg_app <- shinytest2::AppDriver$new(app_dir = "my/shiny/app")
bg_app_url <- bg_app$get_url()
app<- shinytest2::AppDriver$new(paste0(bg_app_url, "?_inputs_&n=10"))
```

With this not being too terrible to have the user implement, (for now) I'm going to just make it a FAQ entry with your solution.

**Comment by schloerke** [comment]
The issues with `_inputs_` != `_inputs=` has been addressed in #222
